src/main/java/javaslang/collection/HashMap.java

```diff
@@ -6,434 +6,74 @@
 package javaslang.collection;
 
 import javaslang.Tuple2;
-import javaslang.control.None;
-import javaslang.control.Option;
 
-import java.io.Serializable;
-import java.util.Objects;
+import java.util.Iterator;
 
 /**
  * A {@code HashMap} implementation based on a
  * <a href="https://en.wikipedia.org/wiki/Hash_array_mapped_trie">Hash array mapped trie (HAMT)</a>.
  */
-public interface HashMap<K, V> extends Map<K, V> {
+class HashMap<K, V> implements Map<K, V> {
```

Comment:
Thanks for being uncomplaining regarding all my commits. I bet you had some merge conflicts the last days... It is under heavy development :-)